internal/wrphandlers/qos/priority_queue.go

```diff
@@ -51,34 +64,69 @@ func (pq *priorityQueue) Enqueue(msg wrp.Message) error {
 	return nil
 }
 
+// trim removes messages with the lowest QualityOfService (taking `prioritizeOldest` into account)
+// until the queue no longer violates `maxQueueSize“.
 func (pq *priorityQueue) trim() {
```

Comment:
Introduced this since we described a desire to keep messages with higher qos value taking the age of the message into account as well